.github/workflows/test.yml

```diff
@@ -1,12 +1,16 @@
-name: Test
+name: Core Repo Tests
 
 on:
   push:
     branches-ignore:
     - 'release/*'
   pull_request:
 env:
-  CONTRIB_REPO_SHA: 5c9e043d6921550d82668788e3758a733fb11cb8
+  # Set variable to 'master' if your change will not affect Contrib.
+  # Otherwise, set variable to the commit of your branch on
+  # opentelemetry-python-contrib which is compatible with these Core repo
+  # changes.
+  CONTRIB_REPO_SHA: master
```

Comment:
I think it's fine to keep it as `master` because this one line specifically is going to change quite a lot from here until GA while cross-repo PRs are being made.

I also think it just reads nicer personally? This is proof that the current code on Core passed tests on the `master` branch of Contrib, no need to go search the commit.

Even if something on Contrib breaks, and no one notices it, then someone on Core will run tests and see that the tests fail. Then they can flag this on Gitter and that breaking change can be fixed/reverted. What's nice is that this way hopefully the Core developer will be able to work with working code more quickly instead of falsely assuming everything is okay because the `CONTIRB_REPO_SHA` is locked to before the bug in Contrib was introduced.

Again it should just be treated as some **default good** case. If everything is good between the two repos, adding `master` to their workflows shouldn't be a problem. If you can't add `master` that should signal a problem.

Let me know what you think? The important thing is that it doesn't block either the Core or Contrib developer as far as I can tell.

Reply:
It's definitely a pain to keep bumping the commit. How does the contrib repo know which SHA of the core repo to use when it builds? I would be in favor of having just the contrib track a specific core commit (since core can go ahead of contrib freely) and leaving this one to master.

> Again it should just be treated as some **default good** case. If everything is good between the two repos, adding `master` to their workflows shouldn't be a problem. If you can't add `master` that should signal a problem.

I agree, but I think it would be bad if both repos track master. If you look through the history or blame of the contrib repo, you need to know which version of core it was using as its dependency to understand the code or re-run the tests.

Reply:
I see your point about being able to go through the history or blame, but they alone won't tell you if the tests passed. If instead we check the GitHub actions test run for the last "passing" test run, the Checkout Action actually resolves the `master` SHA to a commit and tells you which one it was, as seen on line 459.

![image](https://user-images.githubusercontent.com/23139455/99424990-2af8b280-28b7-11eb-803a-3c3530731483.png)

> I would be in favor of having just the contrib track a specific core commit (since core can go ahead of contrib freely) and leaving this one to master.

Contrib is already tracking a specific Core commit

https://github.com/open-telemetry/opentelemetry-python-contrib/blob/master/.github/workflows/test.yml#L9

And like you mentioned this PR points to `master`, so this should be the case you mentioned.

But I'm almost thinking it should be the reverse if we want Core to move freely? In the reverse, Core can move freely because its tied to a commit it knows it works with. And Contrib would be the one that has to adapt if something gets merged into Core `master`.

I'm okay with leaving it like this and iterating based on our experience with it if you are.

Let me know what you think!

Reply:
> But I'm almost thinking it should be the reverse if we want Core to move freely? In the reverse, Core can move freely because its tied to a commit it knows it works with. And Contrib would be the one that has to adapt if something gets merged into Core `master`.

I think we should maintain the one-way dependency that contrib packages are depending on the core packages. The core packages always "work" independent of contrib, because core packages do not depend on contrib.

> Contrib is already tracking a specific Core commit

Awesome, then LGTM 🚢

Reply:
> The core packages always "work" independent of contrib

Yes you're definitely right that Core packages work independent of Contrib!

> , because core packages do not depend on contrib.

But this PR is adding Contrib Tests which do depend on Contrib. And we're saying that these tests should take the changes in whatever current Core PR and test them against the `master` of Contrib. So depending on which commit on Contrib it is, the tests may or may not pass. But maybe that's what you meant by this comment:

> I think we should maintain the one-way dependency that contrib packages are depending on the core packages.

Either way I'm good with it if you are 😄